packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/service/storage_s3_service_impl.dart

```diff
@@ -73,7 +75,7 @@ class StorageS3Service {
   final AWSCredentialScope _signerScope;
   final AWSSigV4Signer _awsSigV4Signer;
 
-  /// Takes in input from [AmplifyStorageS3Dart.list] to compose a
+  /// Takes in input from [AmplifyStorageS3Dart] `list` API to compose a
```

Comment:
The benefit of the other way is that it links to the `list` docs when published to pub

Reply:
OK, yea changed it in this PR, as `AmplifyStorageS3Dart` became the stub, which doesn't have the API directly by itself. `AmplifyStorageS3Dart.list` becomes "not defined". Should replace with [AmplifyStorageS3DartBase.list]?

Can probably add "unimplemented" method to the stub, but maybe an overkill? And this doc is internal, not developers facing.

Reply:
I hate to make you rework it but I need to strongly push back on having a base class for the Dart plugin. I really believe dependency injection is a more scalable and robust solution.

Reply:
Let's chat on Monday about any limitations you're experiencing with DI and see what we can do to fix it. I think tighter integration of `DependencyManager` into the Amplify plugin system will help alleviate some of the current pains around it.